### PR TITLE
feat: add TypeSpec tsp-server support

### PR DESCRIPTION
**Pull request by jeremymeng**
TypeSpec (https://github.com/microsoft/typespec) is a language for defining cloud service APIs and shapes. TypeSpec is a highly extensible language with primitives that can describe API shapes common among REST, OpenAPI, gRPC, and other protocols.

### CODE REVIEW

**Comment by jcs090218** [comment]
You can use this PR #4549 as a reference. :)

**Comment by jeremymeng** [comment]
Updated changelog and added doc entry

File: docs/lsp-clients.json [hunk 1]
```diff
@@ -496,6 +496,15 @@
     "lsp-install-server": "jsts-ls",
     "debugger": "Yes (Firefox/Chrome)"
   },
+  {
+    "name": "typespec",
+    "full-name": "TypeSpec",
+    "server-name": "tsp server",
+    "server-url": https://github.com/microsoft/typespec/tree/main/packages/compiler",
```
Comment (kiennq) [inline docs/lsp-clients.json:503]: it seems that a `"` is missing at url beginning?

Comment (jeremymeng) [reply docs/lsp-clients.json:503]: Good catch! Not sure how I missed that.

Comment (jeremymeng) [inline docs/lsp-clients.json:503]:
```suggestion
    "server-url": "https://github.com/microsoft/typespec/tree/main/packages/compiler",
```

**Comment by jeremymeng** [comment]
@jcs090218 @kiennq / maintainers any other feedback for improvements?

**Review by jcs090218 — APPROVED** [review]

**Pull request merged**